src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -163,7 +169,48 @@ public static async Task<Stream> CopyRemoteStreamLocally(Stream stream, PathSett
 
                 throw;
             }
+        }
+
+        private static void AssertFreeSpace(Stream stream, string filePath)
+        {
+            long streamLength;
+
+            try
+            {
+                streamLength = stream.Length;
+            }
+            catch (NotSupportedException)
+            {
+                // nothing we can do
+                return;
+            }
+
+            var driveName = DiskUtils.GetDriveInfo(filePath, DriveInfo.GetDrives(), out string _).DriveName;
+            if (driveName == null)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failed to get drive name for file '{filePath}'");
+
+                return;
+            }
+
+            var spaceInfo = DiskUtils.GetDiskSpaceInfo(driveName);
+            if (spaceInfo == null)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failed to get space info for drive '{driveName}'");
+
+                return;
+            }
+
+            // + we need to download the snapshot
+            // + we need as much space to restore it
```

Comment:
That isn't true. You need to download it, but you may restore on a different location